lib/puppet/provider/package/rpm.rb

```diff
@@ -313,7 +314,24 @@ def insync?(is)
     return false if [:purged, :absent].include?(is)
     return false if is.include?(self.class::MULTIVERSION_SEPARATOR) && !@resource[:install_only]
 
-    should = resource[:ensure]
+    should = @resource[:ensure]
+    if should =~ /[<>=~]/
```

Comment:
after chatting we concluded that we shouldn't rely on specific characters here, but on the String class itself
```suggestion
    if should.is_a?(String)
```